crates/matrix-sdk/src/sliding_sync/list/request_generator.rs

```diff
@@ -154,7 +152,7 @@ pub(super) fn create_range(
     desired_size: u32,
     maximum_number_of_rooms_to_fetch: Option<u32>,
     maximum_number_of_rooms: Option<u32>,
```

Comment:
I wonder if we should replace `u32` by `Bound` on those places. Maybe as another PR.